### DISCUSSION TITLE
Ways to donate?

### DISCUSSION THREAD

**Discussion by CRImier** [roc-streaming/roc-toolkit#395] [Discuss]
Hi! roc is the first solution that works perfectly for me, streaming to a Pi Zero. Is there any way to support the project monetarily - or, perhaps, with hardware?

Cheers!
Arsenijs

**Comment by gavv** [comment]
Hi! Thanks for you interest and good to know the project was helpful to you.

Currently there is no way to make donations. You are the first who's asking, but I also was thinking about it for a while, so this is probably a good reason to start a discussion.

**Comment by gavv** [comment]
I see four questions to be answered:

1. Ways to spend donations
2. What kinds of donation to expect
3. What platform to use
4. How to distribute donations between members

## Ways to spend donations

I see the following options, in order of preference (in my opinion):

1. **Compensate for minor expenses**

   Right now there are *no* expenses. However in nearest future I'm planning to buy a domain and Google Play account for Roc Droid (regardless of presence of donations). In future we may also need Apple Developer account and maybe hardware for CI, but currently I expect expenses to be very low.

2. **Pay maintainers full time or part time**

   Speaking for myself, I'm currently freelancing to earn money and spend the rest time on my open source projects, mostly Roc. Thus, even small salary will work for me: I'll be able to spend a bit less time freelancing and a bit more time on Roc.

3. **Pay outsiders**

   Some examples: participate in student scholarship programs (like suggested earlier by @Asalle); participate in bug bounty programs; put bounties on some important features; hire a professional developer to implement specific feature (or e.g. designer for mobile app).

## What kinds of donation to expect

I see two sources:

* Donations **from users**
* Donations **from companies**

I don't think that donations from users may be very high in the nearest future. I expect that they will be able to cover only minor expenses and maybe a part-time salary. However, even this may be helpful and of course would be very motivational, at least for me.

I would assume that donations work really well only when big companies want to donate. I'm not aware if Roc is currently used in any company. I think this will become realistic a few releases later, when we'll implement some essential features. So far Roc is still more like a geek project. Though, we anyway can just start collecting donations and see what happens.

There are also two kinds of donations:

* **Recurrent**
* **One-time**

Some platforms support only recurrent donations.

## What platform to use

I'm aware of a few:

* [**Liberpay**](https://liberapay.com/)

    People may contribute recurrent donations to an individual or a team. One-time donations are not supported. In case of team account, donations are distributed between team members using a weighted round-robin scheme :)

    They don't manage a bank account for you, transactions are just redirected from a contributor directly to one of the team members (a new member is selected for each transaction).

    No fees except standard payment fees.

    Top teams: https://liberapay.com/explore/teams

* [**Open Collective**](https://opencollective.com/)

    People may contribute both recurrent or one-time donations. Donations are accumulated on a team account. Team members can then request money by "submitting expense", which should be approved or rejected by team admin. All contributions and expenses are public.

    They can manage a bank account for you. When expense is approved, the money is transferred from team account to you.

    The platform has its own fees in addition to standard payment fees.

    Top teams: https://opencollective.com/discover?show=open%20source

    A rather successful media-related project: https://opencollective.com/obsproject

* [**Patreon**](https://www.patreon.com/) and [**GitHub Sponsors**](https://github.com/sponsors)

    As far as I know, both work similarly:

     * people may contribute only recurrent donations
     * donation goes to the account of your team
     * it's up to you how to distribute money between team members; the platform does not help with it

    AFAIK, GitHub Sponsors does not work in my country currently.

* **Issue-oriented platforms**

    There are also services like [**Bountysource**](https://www.bountysource.com/) and [**Issuehunt**](https://issuehunt.io/), which allow to collect donations for individual issues.

## How to distribute money between members

The platforms above suggest four ways to do it:

* **Liberpay: redirect transactions to different members using weighted round-robin algorithm**

    Pros:

    * Easy! Every member that would like to receive donations can just register and then will automatically receive money from time to time.

    Cons:

    * Every member should be assigned a weight, i.e. a number, and it's totally unclear how to select that number. E.g. blindly counting number of commits or SLOCs is stupid.

* **Open Collective: accumulate money on team account; each team member request money when needed**

    Pros:

    * Flexible and transparent.

    Cons:

    * Requires some initiative from each member.

* **Patreon and GitHub: single bank account; distribute money by yourself**

    Pros:

    * Flexible.

    Cons:

    * Not transparent.
    * Additional burden. I wouldn't like to do this accounting by myself :)

* **Issue-oriented platforms: get money per-issue**

    Pros:

    * Users can donate to specific features.

    Cons:

    * Currently our project is in early stage of development and most issues come from our own roadmap. The project is not ready yet to implement any requested feature: some basis should be implemented first. E.g., Opus support is highly wanted by users, but it requires RTSP support first, which in turn requires some changes in core (which we're currently working on).

**Comment by gavv** [comment]
Now, here is my **personal** opinion on the above topics.

1. **Ways to spend donations**

    At this point, when the project is in early stage of development, I'd prefer to spend money to pay maintainers full time or part time.

    If someone in team except me can benefit from such donations and spend more time on Roc, I'd prefer to pay money to them rather than me because I'm already able to spend almost 50% of my time on Roc and what we really need is more people.

2. **What kinds of donation to expect**

    I think we should support both recurrent and one-time donations since it would be more convenient to users.

    I don't think we can expect donations from companies in the nearest future, but we should choose a platform that will make it feasible and convenient, since most likely it's the most important source of donations.

3. **What platform to use**

    I don't think per-issue bounties is a good idea for us.

    I also wouldn't like to bother with accounting and would prefer that the platform could distribute money between members somehow and that this process would be transparent.

    So I'd choose between Liberpay and Open Collective.

    It seems that projects on Open Collective raise much more money than on Liberpay (even the same project, e.g. F-Droid). Open Collective also supports one-time donations and is more flexible.

    Thus, I think Open Collective would be the best choice among the alternatives I listed above.

4. **How to distribute donations between members**

    If we decide to use Open Collective and pay our members full- or part-time, then each member that would like to be paid should create a request on Open Collective, something like "full-/part-time work on feature X for N weeks" or even just "full-/part-time work for N weeks".

    Since currently we're a very small community, I think we can just discuss such things in the mailing list without unnecessary bureaucracy.

**Comment by gavv** [comment]
What do others think on the four topics above?

In particular, does anybody have a desire to be paid full- or part- time to work on this project?

@baranovmv @dshil @Asalle @MatteoArella @ortex

@alexandremgo Is helping us with the release, so also casting him!

@CRImier from the user perspective, what way of donations would you prefer and how would you expect the money to be spent?

**Comment by Asalle** [comment]
Hey all,
Amazing idea and I'm happy to see roc grow!
I personally would not like to be paid. My contribution not that big and I
would prefer the funds to be used to attract more people.
Also, I am not a freelancer and I don't know how to report this kind of
income to tax office 🤔

Regarding the ways to gather donations, I think we should provide more ways
to give - including github sponsorships, eliminating all the boundaries
between us and sponsors.

BR,
Asa

**Comment by ortex** [comment]
Hi, all!
My contributions are quite small and I think I'll continue to do some stuff from time to time, but I don't want to be paid. 

P.s: I'm glad to see that someone want to support the project 👍

**Comment by dshil** [comment]
Hi @gavv,

Before I will dig into discussions of 4 topics you specified above I want to clarify a few things. For now you are only a single core maintainer of the project. I know that there are other people who help with other parts of the project but without you I am not sure that Roc will continue to grow unless some company will hire you to continue working on Roc inside the company. Thus,  I am more interested in supporting you directly to ensure Roc and other your projects will continue to grow. I want you to think about dividing donations personally to your self and to Roc organization. If you will have personal patreon or github sponsors account me and other people can support you directly (as a sigh of respect and support).

At the end all people interested in donation may also have github sponsors or patreon account and other people may support them also without thinking about supporting the entire organization. 

As a result I think it is too early to think about all these topics unless some company will have some interest to support Roc organization and not only you.

And now the time to answer all your 4 questions.

> Ways to spend donations

- Pay people (most important);

- Infrastructure payment but probably we can minimize them too because
  Microsoft, DO (digital ocean)  and others may provide some kind of support
  to the open source projects (not sure).

> What kinds of donation to expect

- Company (donation and paid feature requests);
- Bounty programs (paid feature requests);
- Personal (some people still may want to support the entire organization).

> What platform to use
> So I'd choose between Liberpay and Open Collective.

I think Open Collective is a good choice.

> How to distribute donations between members

Core team should be responsible for it. First things first. After the infrastructure payments and people payments will be done the remaining can be used by requests from the team members.

> AFAIK, GitHub Sponsors does not work in my country currently.

Does it mean that we can support the project but you won't be able to withdraw
some money for your self, correct?

**Comment by gavv** [comment]
@Asalle @ortex 

Thanks for responses!

@dshil

Thanks for response too and thanks for your words!

I see you're talking about the C++ part, and in this case - yes, I'm the only active maintainer these days. But fortunately the project becomes bigger than just the C++ core, we now also have Java and Go bindings and an Android app, which are developed by other people.

I believe, for an OSS project, the most important thing is to grow its community, i.e. attract more people for all parts of the project. Without community, I think it's unlikely that the project can become successful or become backed by companies.

When I say "community", I primarily mean not just casual contributors (which are important too, however!), but people who wish to stay with the project for a longer time and maybe maintain some part of it. Finding such people is the main reason why we have all those help wanted issues.

So what I'm talking about? If we had some money and see a way how to use them to grow the team or involvement of the members, I'd prefer to do it. For example, if someone will have an opportunity to invest more time if being paid. If no one would have such an opportunity, I can always just take the money and spend one more month or week on Roc instead of freelancing.

I see two disadvantages of not having a single account for organization:

* It may be less transparent to the users or companies who want to donate.
* Doesn't allow us to put together many small donations and spend them on some bigger thing, like one month of full-time for a single developer.

Since anyway we don't expect much donations in the nearest future, and Open Collective looks good, maybe just create an account and see how it's going? We can just accumulate those small donations on the account for a while.

Regarding personal account, I actually was thinking about creating it too (in future), though I was more thinking about supporting my other small projects except Roc. I think I can add it too.

Regarding Github Sponsors. Open Collective can create a bank account for you, which will work as a buffer accumulating money. With Github Sponsors, on the other hand, you should provide your own account.

This mean two things: 1) you should do the accounting by yourself (e.g. transfer money to team members), and 2) you need a bank account in the country which is supported by Github Sponsors. Not all countries are supported so far. So if I understand correctly, currently I can't even create a Github Sponsor account, without using tricks like Payoneer.

**Comment by gavv** [comment]
Hi all!

I have created liberapay team: https://liberapay.com/roc-streaming/

If someone work on this project or have such plans, and would like to receive those donations, please let me know! Will be happy to add you. When liberapay team receives donations, they will be distributed among team members. For now, the team contains only me. I don't expect much donations here, but if they turn up, I'll use my share to spend more time on the project.

I also plan to create opencollective account a bit later. If it will receive any donations, they will be accumulated on opencollective account. I think it makes sense to wait until some meaningful amount would be accumulated and then decide how to spend it.

I also created personal accounts for donations (https://liberapay.com/gavv/, https://www.paypal.com/paypalme/victorgaydov). I placed links to them in my personal repositories outside of roc.

**Comment by gavv** [comment]
I've registered project on Open Collective: https://opencollective.com/roc-streaming

**Comment by gavv** [comment]
https://github.com/roc-streaming/roc-go/pull/39
https://github.com/roc-streaming/roc-java/pull/44
https://github.com/roc-streaming/roc-droid/pull/40
